src/components/Markings/components/ShapeEditorSvg.js

```diff
@@ -94,36 +106,90 @@ class ShapeEditorSvg extends Component {
                         shapeIndex: touchedShapeIndex,
                         touchState: touchedSquare
                     })
-                }             
+                } 
+                // If a shape isn't being touched, then we should begin drawing a new shape 
+                else {
+                    const previewShapeDimensions = R.merge(this.state.previewShapeDimensions, {
+                        x: (locationX - INITIAL_PREVIEW_SHAPE_SIDE) * this.props.displayToNativeRatioX,
+                        y: (locationY - INITIAL_PREVIEW_SHAPE_SIDE) * this.props.displayToNativeRatioY,
+                    })
+                    this.setState({
+                        isDrawing: true,
+                        previewShapeDimensions
+                    })
+                }
             },
             onPanResponderMove: (evt, gestureState) => {
-                const { shapeIndex, touchState, shapeRefs } = this.state
+                const { shapeIndex, touchState, shapeRefs, isDrawing } = this.state
+                const { locationX, locationY } = evt.nativeEvent
+                const { dx, dy } = gestureState
                 if (shapeIndex >= 0) {
-                    const { dx, dy } = gestureState;
                     // Because Svgs don't have any way to animate, we have to update their props manually
                     const deltas = calculateShapeChanges(touchState, dx, dy, this.props.displayToNativeRatioX, this.props.displayToNativeRatioY)
-                    shapeRefs[shapeIndex].update(deltas);
+                    const newDimensions = shapeRefs[shapeIndex].update(deltas);
+                    const shapeIsOutOfBounds = isShapeOutOfBounds(newDimensions, {width: this.props.width *this.props.displayToNativeRatioX, height: this.props.height * this.props.displayToNativeRatioY}) && touchState.onlySquare
```

Comment:
What does `onlySquare` reference exactly? I'm sure it's necessary, but I'm curious why based on the naming. Not sure why it needs to be true to determine if out of bounds.

Reply:
`onlySquare` is describing how the user is touching the shape. It means the user is touching the shape but not touching any of the corners. Essentially it means the square shouldn't be resized, just dragged. 

The reason we check it here is because the delete action only happens on drag out of bounds, not resize out of bounds. 

Let me change around some of the naming to make this more clear.